opencti-platform/opencti-front/src/private/components/data/connectors/Connector.jsx

```diff
@@ -241,6 +251,30 @@ const ConnectorComponent = ({ connector, relay }) => {
       filterGroups: [],
     },
   };
+  const { me } = useAuth();
+  const theme = useTheme();
+  const userCapabilities = (me.capabilities ?? []).map((c) => c.name);
+  const userHasSettingsCapability = userCapabilities.includes(SETTINGS_SETACCESSES) || userCapabilities.includes(BYPASS);
+
+  const checkLastRunExistingInState = () => {
+    let lastRun = false;
+    const connectorStateConverted = JSON.parse(connector.connector_state);
+    if (Object.hasOwn(connectorStateConverted, 'last_run')) {
+      lastRun = true;
+      return lastRun;
+    }
+    return lastRun;
+  };
+  const getLastRunConverted = () => {
+    const connectorStateConverted = JSON.parse(connector.connector_state);
```

Comment:
I don't think it's a good idea to JSON.parse `connector.connector_state` in methods, it should be done once only, not at every call of these methods.